pkg/operator/controller/ingress/controller.go

```diff
@@ -189,13 +197,13 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 // fields.  Returns an error value, which will have a non-nil value of type
 // admissionRejection if the ingresscontroller was rejected, or a non-nil
 // value of a different type if the ingresscontroller could not be processed.
-func (r *reconciler) admit(current *operatorv1.IngressController, ingressConfig *configv1.Ingress, infraConfig *configv1.Infrastructure) error {
+func (r *reconciler) admit(current *operatorv1.IngressController, ingressConfig *configv1.Ingress, infraConfig *configv1.Infrastructure, tlsProfileSpec *configv1.TLSProfileSpec) error {
 	updated := current.DeepCopy()
 
 	setDefaultDomain(updated, ingressConfig)
 	setDefaultPublishingStrategy(updated, infraConfig)
 
-	if err := r.validate(updated); err != nil {
+	if err := r.validate(updated, tlsProfileSpec); err != nil {
```

Comment:
It's not clear to me why tlsProfile is being treated like a special case compared to domain and publishing strategy throughout. I was expecting the new argument to this function to be `apiConfig` and a new `setDefaultTLSProfile()` function or something. Was there a reason the pattern couldn't be applied to this field?

Reply:
(see also: `validate()` which is now unusually special-cased for tlsProfile)

Reply:
> It's not clear to me why tlsProfile is being treated like a special case compared to domain and publishing strategy throughout. I was expecting the new argument to this function to be `apiConfig` and a new `setDefaultTLSProfile()` function or something. Was there a reason the pattern couldn't be applied to this field?

Because we decided that `status.tlsProfile` should not be set till pods are at level.  What would `setDefaultTLSProfile` do if not set `status.tlsProfile`, or what would `validate` use to obtain the TLS profile if not a parameter value?

Reply:
>Because we decided that status.tlsProfile should not be set till pods are at level. What would setDefaultTLSProfile do if not set status.tlsProfile, or what would validate use to obtain the TLS profile if not a parameter value?

Is the important difference here between TLS and domain/publishing that TLS is mutable? That is, it's okay to persist status for domain/publishing in advance of rollout because there's simply no opportunity for divergence between spec/status for immutable fields?

Reply:
> Is the important difference here between TLS and domain/publishing that TLS is mutable?

Basically yeah (although someone could argue that _strictly speaking_ we shouldn't set `status.domain` or `status.endpointPublishingStrategy` till pods are up either).